core/src/mindustry/world/blocks/storage/Unloader.java

```diff
@@ -85,9 +85,23 @@ public class UnloaderBuild extends Building{
         public int[] lastUsed;
 
         protected final Comparator<ContainerStat> comparator = Structs.comps(
+            //sort so it gives priority for blocks that can only either recieve or give (not both), and then by load, and then by last use
+            //highest = unload from, lowest = unload to
+
             Structs.comps(
-                Structs.comparingBool(e -> e.building.block.highUnloadPriority && !e.canLoad),
-                Structs.comparingBool(e -> e.canUnload) // && !e.canLoad
+                Structs.comparingBool(e -> e.building.block.highUnloadPriority && !e.canLoad), //stackConveyors and Storage
+                Structs.comps(
+                    Structs.comparingBool(e -> {
+                        boolean i = e.canUnload && !e.canLoad;
+                        //if(i){ System.out.println(e);}
```

Comment:
The debug / extra return code is messy, it's best if you remove it here.

Reply:
wait i forgot to remove that lol